rlbot_gui/gui/js/story-mode-start.js

```diff
@@ -36,11 +52,29 @@ export default {
         return {
             form: {
                 teamname: '',
-                teamcolor: 0
+                teamcolor: 0,
+                story_id: 'default',
+                custom_story: {
+                    challenge: '',
+                    bots: ''
+                }
             },
+            storyIdOptions: [
+                { value: "easy", text: "Easy"},
+                { value: "default", text: "Default"},
+                { value: "custom", text: "Custom"}
+            ]
         };
     },
     methods: {
+        pickFile: async function(event) {
+            let field = event.target.value;
+
+            let path = await eel.pick_location(false, 'JSON files (*.json)')(); // is_folder=False
```

Comment:
You can potentially use the browser's file picker here instead of the python one, so it would pop up in the foreground.

We only use python for file picking so far because we needed to know a path in the file system, but you could just throw json blobs around for this, right?

Might need to store a redundant copy of the blobs locally, but that would insulate them from getting changed while somebody's story is in progress.

Reply:
This is an interesting idea.
My concern is about the workflow of someone who might be iterating on their version of `challenges.json`. For testing any change, they will have to delete their save, start a new save with the new file and then beat enough matches to see the impact of their change. With the current implementation, they just have to restart RLBotGUI to see their change take effect.

Reply:
RLBotGUI could store something like active-challenges.json in a safe location, e.g. in the content folder https://github.com/RLBot/RLBotGUI/blob/2e4112fa40660ee410c660c8361f8441cd32ad65/rlbot_gui/gui.py#L431

and then store that file path in the story save. Developers could modify that file in place to iterate.

1. Browser file picker
1. Transmit over eel as json
1. Python receives as dict
1. Convert to json again
1. Write to file in content folder
1. Use file path as you did before

And then maybe in dev mode, the GUI could expose the file path of the currently active challenges.

Reply:
Anyways I shouldn't have blocked this PR on it, I'm gonna go ahead and merge / deploy. If you like my proposal then I might implement myself. Can't wait for people to mess with their own json files 😎

Reply:
That sounds reasonable to me! Would be good if "dev mode" didn't actually need any developer experience, so folks can find the paths without doing anything too technical.